src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_NativeInteger.cs

```diff
@@ -916,6 +916,33 @@ public void AttributeFieldExists()
             });
         }
 
+        [Theory]
+        [InlineData("C<dynamic, nint, System.IntPtr>", "C<dynamic, System.IntPtr, System.IntPtr>", "False, False, True, False")]
+        [InlineData("C<dynamic, nuint, System.UIntPtr>", "C<dynamic, System.UIntPtr, System.UIntPtr>", "False, False, True, False")]
+        [InlineData("C<T, nint, System.IntPtr>", "C<T, System.IntPtr, System.IntPtr>", "False, False, True, False")]
+        [InlineData("C<T, nuint, System.UIntPtr>", "C<T, System.UIntPtr, System.UIntPtr>", "False, False, True, False")]
+        public void NestedNativeIntegerWithPrecedingType(string sourceType, string metadataType, string expectedAttribute)
+        {
+            var comp = CompileAndVerify($@"
+class C<T, U, V>
+{{
+    public {sourceType} F;
+}}
+", options: TestOptions.ReleaseDll, parseOptions: TestOptions.RegularPreview, symbolValidator: symbolValidator);
+
+            void symbolValidator(ModuleSymbol module)
```

Comment:
>void [](start = 12, length = 4)

Consider making `static` or inline in `CompileAndVerify` to match the other tests.